chrome/content/zotero/xpcom/proxy.js

```diff
@@ -417,26 +425,37 @@ Zotero.Proxies = new function() {
 	  * Show a proxy-related notification
 	  * @param	{Array}		bw			output of _getBrowserWindow
 	  * @param	{String}	label		notification text
-	  * @param	{String}	button		button text ("settings" or "enable")
-	  * @param	{Function}	callback	callback to be executed if button is pressed
+	  * @param	{Array}	  buttons		dicts of button label ("settings", "enable", or "hide") and associated callback
 	  */
-	 function _showNotification(bw, label, button, callback) {
+	 function _showNotification(bw, label, buttons) {
 	 	var browser = bw[0];
 	 	var window = bw[1];
-	 	
+
+		buttons = buttons.map(function(button) {
```

Comment:
This is in the spirit of how the code worked before I started messing with it; I personally think it does cut down on having a bunch of redundant code in places that are already pretty verbose.